qutebrowser/mainwindow/mainwindow.py

```diff
@@ -169,6 +169,9 @@ def __init__(self, *, private, geometry=None, parent=None):
         objreg.register('message-bridge', message_bridge, scope='window',
                         window=self.win_id)
 
+        if config.val.window.hide_decoration:
+            window_flags = Qt.CustomizeWindowHint | Qt.NoDropShadowWindowHint
+            self.setWindowFlags(Qt.Window | window_flags)
```

Comment:
This should be handled in `_on_config_changed` too if possible so it can be set without requiring a restart. Not sure if it's possible to just change the flags after the window has been shown though - but maybe a `self.hide()` and `self.show()` helps.